File: tourneyAPI/Models/Entities/Characters/Luigi.cs
namespace Entities;

using Enums;

// Defines the competitive profile metadata for this playable character.
public class Luigi : Character
{
    // Initializes this character's default competitive attributes for matchmaking and tier logic.
    public Luigi()
    {
        Id = CharacterId.Luigi;
        characterName = CharacterName.LUIGI;
        archetype = Archetype.HALF_GRAPPLER;
        fallSpeed = FallSpeed.FLOATY;
        weightClass = WeightClass.LIGHTWEIGHT;
        tierPlacement = TierPlacement.A;
    }

}

File: tourneyClient/src/models/entities/Characters/Luigi.ts
import { CharacterId } from "@/models/Enums/CharacterId";
import { Character } from "../Character"
import { CharacterName } from "@/models/Enums/CharacterName";
import { Archetype } from "@/models/Enums/Archetype";
import { FallSpeed } from "@/models/Enums/FallSpeed";
import { WeightClass } from "@/models/Enums/WeightClass";
import { TierPlacement } from "@/models/Enums/TierPlacement";

export const Luigi: Character =
{
    id: CharacterId.Luigi,
    characterName: CharacterName.LUIGI,
    archetype: Archetype.HALF_GRAPPLER,
    fallSpeed: FallSpeed.FLOATY,
    weightClass: WeightClass.LIGHTWEIGHT,
    tierPlacement: TierPlacement.A
}

export default Luigi; 